examples/unstructured_2d_dgsem/elixir_shallowwater_twolayer_dam_break.jl

```diff
@@ -0,0 +1,140 @@
+
+# This initial condition will be overwritten with the true discontinuous condition from 
+# initial_condition_discontinuous_dam_break
+function initial_condition_dam_break(x, t,equations::ShallowWaterTwoLayerEquations2D)
+  if x[1] < sqrt(2)/2
+    H1 = 1.0
+    H2 = 0.6
+    b  = 0.1
+  else
+    H1 = 0.9
+    H2 = 0.5
+    b  = 0.0
+  end
+
+  v1 = 0.0
+  w1 = 0.0
+  v2 = 0.0
+  w2 = 0.0
+  return prim2cons(SVector(H1, v1, w1, H2, v2, w2, b), equations)
+end
+
+initial_condition = initial_condition_dam_break
+
+boundary_condition_constant = BoundaryConditionDirichlet(initial_condition_dam_break)
+
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_wintermeyer_etal, flux_nonconservative_wintermeyer_etal)
+surface_flux= (flux_fjordholm_etal, flux_nonconservative_fjordholm_etal)
+solver = DGSEM(polydeg=6, surface_flux=surface_flux,
+               volume_integral=VolumeIntegralFluxDifferencing(volume_flux))
+
+###############################################################################
+# Get the unstructured quad mesh from a file (downloads the file if not available locally)
+default_mesh_file = joinpath(@__DIR__, "mesh_alfven_wave_with_twist_and_flip.mesh")
+isfile(default_mesh_file) || download("https://gist.githubusercontent.com/andrewwinters5000/8f8cd23df27fcd494553f2a89f3c1ba4/raw/85e3c8d976bbe57ca3d559d653087b0889535295/mesh_alfven_wave_with_twist_and_flip.mesh",
+                                      default_mesh_file)
+mesh_file = default_mesh_file
+
+mesh = UnstructuredMesh2D(mesh_file, periodicity=false)
+
+# Boundary conditions
+boundary_condition = Dict(:Top    => boundary_condition_slip_wall,
+                          :Left   => boundary_condition_slip_wall,
+                          :Right  => boundary_condition_slip_wall,
+                          :Bottom => boundary_condition_slip_wall)
+
+# Create the semi discretization object
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, 
+                                    solver, boundary_conditions=boundary_condition)
+
+###############################################################################
+# ODE solver
+
+tspan = (0.0, 0.5)
+ode = semidiscretize(semi, tspan)
+
+###############################################################################
+# Workaround to set a discontinuous bottom topography and initial condition for debugging and testing.
+
+# alternative version of the initial conditinon used to setup a truly discontinuous
+# test case and initial condtion.
+# In contrast to the usual signature of initial conditions, this one get passed the
+# `element_id` explicitly. In particular, this initial conditions works as intended
+# only for the specific mesh loaded above!
+
+function initial_condition_discontinuous_dam_break(x, t, element_id,equations::ShallowWaterTwoLayerEquations2D)
```

Comment:
```suggestion
function initial_condition_discontinuous_dam_break(x, t, element_id, equations::ShallowWaterTwoLayerEquations2D)
```

Reply:
Done in 7c9fde8356a8fc13cdea0b9313579ed924402a41